pysatMadrigal/tests/test_utils_coords.py

```diff
@@ -0,0 +1,443 @@
+"""
+tests the pysat coords area
+"""
+import numpy as np
+
+from pysatMadrigal.utils import coords
+
+
+class TestGeodeticGeocentric():
+
+    def setup(self):
+        """Runs before every method to create a clean testing setup."""
+        arr = np.ones(shape=(10,), dtype=float)
+        self.val = {'lat': 45.0, 'lon': 8.0, 'az': 52.0, 'el': 63.0}
+        self.arr = {}
+        for key in self.val.keys():
+            self.arr[key] = self.val[key] * arr
+
+    def teardown(self):
+        """Runs after every method to clean up previous testing."""
+        del self.val, self.arr,
+
+    def test_geodetic_to_geocentric_single(self):
+        """Test conversion from geodetic to geocentric coordinates"""
+
+        lat, lon, rad = coords.geodetic_to_geocentric(self.val['lat'],
+                                                      lon_in=self.val['lon'])
+
+        assert abs(lat - 44.807576784018046) < 1.0e-6
+        assert abs(lon - 8.0) < 1.0e-6
+        assert abs(rad - 6367.489543863465) < 1.0e-6
+
+    def test_geocentric_to_geodetic_single(self):
+        """Test conversion from geocentric to geodetic coordinates"""
+
+        lat, lon, rad = coords.geodetic_to_geocentric(self.val['lat'],
+                                                      lon_in=self.val['lon'],
+                                                      inverse=True)
+
+        assert abs(lat - 45.192423215981954) < 1.0e-6
+        assert abs(lon - 8.0) < 1.0e-6
+        assert abs(rad - 6367.345908499981) < 1.0e-6
+
+    def test_geodetic_to_geocentric_mult(self):
```

Comment:
These two can be combined with parametrization, and then expanded to test list input.  This also applies to the other single/mult pairs.